apps/api/src/main/java/dk/treecreate/api/newsletter/NewsletterController.java

```diff
@@ -0,0 +1,109 @@
+package dk.treecreate.api.newsletter;
+
+import dk.treecreate.api.authentication.services.AuthUserService;
+import dk.treecreate.api.exceptionhandling.ResourceNotFoundException;
+import dk.treecreate.api.newsletter.dto.GetNewslettersResponse;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.v3.oas.annotations.Operation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.UUID;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("newsletter")
+@Api(tags = {"newsletter"})
+public class NewsletterController
+{
+    @Autowired
+    NewsletterRepository newsletterRepository;
+    @Autowired
+    AuthUserService authUserService;
+
+    @GetMapping()
+    @Operation(summary = "Get all newsletters")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "A list of newsletters",
+            response = GetNewslettersResponse.class)})
+    @PreAuthorize("hasRole('DEVELOPER') or hasRole('ADMIN')")
+    public List<Newsletter> getNewsletters()
+    {
+        return newsletterRepository.findAll();
+    }
+
+    @GetMapping("me")
+    @Operation(
+        summary = "Get a newsletter associated (via email) with the currently authenticated user")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Newsletter information",
+            response = Newsletter.class),
+        @ApiResponse(code = 404, message = "Newsletter not found")
+    })
+    @PreAuthorize("hasRole('USER') or hasRole('DEVELOPER') or hasRole('ADMIN')")
+    public Newsletter getAssociatedNewsletter()
+    {
+        var userDetails = authUserService.getCurrentlyAuthenticatedUser();
+        return newsletterRepository.findByEmail(userDetails.getEmail())
+            .orElseThrow(() -> new ResourceNotFoundException("Newsletter not found"));
+    }
```

Comment:
This works well now, also because we will need the _getAssociatedNewsletter()_ such few times while using the page, but might it not be better to add a field in the IUser with *isSubscribed*? Maybe thats not great because we would have to change the user object if someone unsubscribes. Just a thought.